tutorial/server/axum/assets/auth.js

```diff
@@ -0,0 +1,93 @@
+function register () {
+    let username = document.getElementById('username').value;
+    if (username === "") {
+        alert("Please enter a username");
+        return;
+    }
+
+    fetch('http://localhost:8080/register_start/' + username, {
+        method: 'POST'
+    })
+    .then(response => response.json() )
+    .then(credentialCreationOptions => {
+        credentialCreationOptions.publicKey.challenge = Base64.toUint8Array(credentialCreationOptions.publicKey.challenge);
+        credentialCreationOptions.publicKey.user.id = Base64.toUint8Array(credentialCreationOptions.publicKey.user.id);
+
+        return navigator.credentials.create({
+            publicKey: credentialCreationOptions.publicKey
+        });
+    })
+    .then((credential) => {
+        fetch('http://localhost:8080/register_finish', {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify({
+                id: credential.id,
+                rawId: Base64.fromUint8Array(new Uint8Array(credential.rawId), true),
+                type: credential.type,
+                response: {
+                    attestationObject: Base64.fromUint8Array(new Uint8Array(credential.response.attestationObject), true),
+                    clientDataJSON: Base64.fromUint8Array(new Uint8Array(credential.response.clientDataJSON), true),
+                },
+            })
+        })
+        .then((response) => {
+            if (response.ok){
+                console.log("Registered!");
```

Comment:
Can success / failure be shown on the page as well?

ie: `document.getElementById('status').innerText = "Registered!";`